BootloaderCorePkg/Tools/BuildUtility.py

```diff
@@ -536,6 +536,20 @@ def copy_expanded_file (src, dst):
     gen_cfg_data ("GENDLT", src, dst)
 
 def gen_config_file (fv_dir, brd_name_override, brd_name, platform_id, pri_key, cfg_db_size, cfg_size, cfg_int, cfg_ext, sign_scheme, hash_type, svn, brd_build_name):
+    # boards using CfgDataDef yaml files with board extension
+    # mapping from brd_build_name to board_extension
+    yaml_board_extension_map = {
+        # ADL series
+        'adln'  : 'Adln',
+        'adlp'  : 'Adlp',
+        'adlps' : 'Adlp',
+        'adls'  : 'Adls',
+        # RPL series
+        'rplp'  : 'Adlp',
+        'rplps' : 'Adlp',
+        'rpls'  : 'Adls',
+    }
```

Comment:
BuildUtility.py is a common tool. I think platform specific settings should come from BoardConfig.py.
how about add below line in the BoardConfig.py?
        self.CFG_DATA_MAIN_FILE   = 'Platform/AlderlakeBoardPkg/CfgData/CfgDataDef_rpls.inf'

Reply:
Will do accordingly. Always thanks for pointing out the design concept